File: 001/b.js
function swap(a, b) {
    [a, b] = [b, a];
  }
  
  function main() {
    let n, x;
    // Lendo os valores de n e x da entrada
    n = parseInt(readline());
    x = parseInt(readline());
    
    let ans = n;
    let a = n - x;
    let b = x;
    
    if (a < b) {
      swap(a, b);
    }
    
    while (b !== 0) {
      let k = Math.floor(a / b);
      ans += k * b * 2;
      let tmp = a % b;
      a = b;
      b = tmp;
    }
    
    console.log(ans - a);
  }
  
  main();

File: 001/b.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long LL;
int main() {
	LL n,x;
	cin >> n >> x;
	LL ans = n;
	
	LL a = n-x, b = x; // a = 3 e b = 2
	if(a<b) swap(a,b);
	while(b!=0) {
		LL k = a / b;
		ans += k * b * 2;
		LL tmp = a % b;
		a = b;
		b = tmp;
		
	}
	cout << ans - a << endl;
	
	
	return 0;
}

